### DISCUSSION TITLE
微博个人时间线首次授权时需要关闭访问控制

### DISCUSSION THREAD

**Discussion by aaro-n** [DIYgod/RSSHub#11566] [Show and tell]
# 目前遇到的问题
RSSHub部署在vercel上，我在日常使用中会启用`ACCESS_KEY`，按照文档申请微博API并配置，RSSHub已经配置好`WEIBO_APP_KEY``WEIBO_APP_SECRET``WEIBO_REDIRECT_URL`，之后访问`https://rsshub.xxx.fun/weibo/timeline/1111?key=222`，会提示访问错误，临时关闭`ACCESS_KEY`，再次访问`https://rsshub.xxx.fun/weibo/timeline/1111`，会显示微博授权界面，授权后再启用`ACCESS_KEY`，可以正常访问`https://rsshub.xxx.fun/weibo/timeline/1111?key=222`，大概可以使用1年，之后要重新进行授权。
# 微博及RSSHub部分设置
微博设置
OAuth2.0 授权设置：`https://rsshub.xxx.fun/weibo/timeline/0`（尝试设置`https://rsshub.xxx.fun/weibo/timeline/0?key=222`,再次访问`https://rsshub.xxx.fun/weibo/timeline/1111?key=222`，没有用，首次访问（授权）还是会出现相同的错误）
RSSHub设置
ACCESS_KEY：`222`
WEIBO_REDIRECT_URL: `https://rsshub.xxx.fun/weibo/timeline/0`（同样尝试设置`https://rsshub.xxx.fun/weibo/timeline/0?key=222`，没有用）
WEIBO_APP_SECRET：`2bbfce171xxx5420433cxxxx`
WEIBO_APP_KEY：`8xxxxx`
# 自己探索
* [设置`用户认证`的用户名和密码](https://docs.rsshub.app/install/#pei-zhi-yong-hu-ren-zheng)，`OAuth2.0 授权设置`和`WEIBO_REDIRECT_URL`设置为`https://user:pass@rsshub.xxx.fun/weibo/timeline/0`，首次授权访问会出错
* [设置`ACCESS_KEY`](https://docs.rsshub.app/install/#pei-zhi-fang-wen-kong-zhi-pei-zhi)，找不到MD5转码的方法
# 临时解决方法
* 先关闭`访问控制`并设置URL，等首次授权后再开启`访问控制`
* 使用白名单即`WHITELIST`设置为`/weibo/timeline/0,/weibo/timeline/微博用户名`，可以在启用`ACCESS_KEY`的情况下直接访问。

**Comment by ZefanHu** [comment]
你好，我这边无法访问微博个人时间线，你能帮我看看哪里出问题了吗？我刷新了好几遍，但是手机微博没有弹出啥授权页面，不知道在哪里授权呢？
![image](https://github.com/user-attachments/assets/de723b1f-65ea-4aee-884f-ee8fcd0ec341)

![image](https://github.com/user-attachments/assets/a2ea08e7-b894-4a18-ba2a-2570d4495a48)

![image](https://github.com/user-attachments/assets/4ca2c389-0bcd-4d8f-95b6-544262a66127)

![image](https://github.com/user-attachments/assets/66f6065f-887b-4615-968a-9988feac707c)

**Reply by ZefanHu** [reply]
稍微查了下微博开发文档，手动访问授权页面，拿微博扫一下弹出的二维码就行。RSSHub没有自动跳转，也许是一个bug。
